### PR TITLE
feat(#43): fakehub start, --port

### LINKED ISSUE #43: args.rs:24-27: Handle port argument We should process the...
The puzzle `13-97a91bb1` from #13 has to be resolved: 

https://github.com/h1alexbel/fakehub/blob/f8bde4ef70b3c73a6cbd9a4570141989eebd5fe3/cli/src/args.rs#L24-L27

The puzzle was created by @h1alexbel on 23-Jun-24. 

Estimate: 15 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by h1alexbel**
@l3r8yJ take a look, please

closes #43

---

## PR-Codex overview
**Focus:** Refactoring CLI to support starting a server with logging.

### Detailed summary
- Added `log::info` for logging.
- Implemented `Server` struct.
- Updated `Args` struct with `Command` enum.
- Added `Start` subcommand for starting the server.
- Updated integration tests for help and start options.
- Added integration test for starting the server.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: cli/tests/integration_test.rs [hunk 1]
```diff
@@ -25,12 +25,48 @@ use anyhow::Result;
...
+    assert!(output.contains("start"));
+    assert!(output.contains("Start the server"));
+    assert!(output.contains("--help"));
+    assert!(output.contains("Print help"));
+    Ok(())
+}
+
+#[test]
+fn outputs_start_opts() -> Result<()> {
+    env!("CARGO_BIN_EXE_cli");
```
Comment (l3r8yJ) [inline cli/tests/integration_test.rs:43]: @h1alexbel btw I'm not sure that this is an required, can you try to remove it?

**Review by l3r8yJ — CHANGES_REQUESTED** [review]
@h1alexbel take a look at my comments below, please

File: cli/tests/integration_test.rs (hunk 1 above)
Comment (h1alexbel) [reply cli/tests/integration_test.rs]: @l3r8yJ removed, test passed.

**Comment by h1alexbel** [comment]
@l3r8yJ take a look, again please

File: cli/tests/integration_test.rs [hunk 2]
```diff
@@ -25,12 +25,47 @@ use anyhow::Result;
...
+
+// @todo #43:30min Enable starts_server integration test.
+//  We should enable this integration test that checks whether server starts or
+//  not. This test should output success message in info!. For now it does not
+//  work. Probably some error with logging configuration. Don't forget to
+//  remove this puzzle.
+#[test]
+#[ignore]
+fn starts_server() -> Result<()> {
+    env!("CARGO_BIN_EXE_cli");
```
Comment (l3r8yJ) [inline cli/tests/integration_test.rs:62]: redundant `env!`

File: cli/src/args.rs [hunk 3]
```diff
@@ -19,18 +19,28 @@
...
 #[derive(Parser, Debug)]
+#[command(name = "fakehub")]
 pub(crate) struct Args {
+    #[command(subcommand)]
+    pub(crate) command: Command,
+}
+
+#[derive(Subcommand, Debug)]
+pub(crate) enum Command {
+    // Start the server
```
Comment (l3r8yJ) [inline cli/src/args.rs:37]:
```suggestion
    /// Start the server
```

File: cli/tests/integration_test.rs [hunk 4]
```diff
@@ -25,12 +25,47 @@ use anyhow::Result;
 use assert_cmd::Command;
 
 #[test]
-fn should_output_help() -> Result<()> {
+fn outputs_help() -> Result<()> {
     env!("CARGO_BIN_EXE_cli");
```
Comment (l3r8yJ) [inline cli/tests/integration_test.rs:29]:
```suggestion
```

**Review by l3r8yJ — CHANGES_REQUESTED** [review]
@h1alexbel take a look, please

**Comment by h1alexbel** [comment]
@l3r8yJ updated

**Review by l3r8yJ — APPROVED** [review]
@h1alexbel looks good to me!

**Pull request merged**